hermes/probe/probe.go

```diff
@@ -33,91 +33,85 @@ import (
-	Journal *journalpb.StateJournal
-
-	// LatencyMetrics stores the api call and probe operation latency for a given target run.
-	// Metrics are stored with additional labels to record operation type and exit status.
-	LatencyMetrics *metrics.Metrics
-}
+const (
+	// TODO(evanSpendlove): Refactor to use constant from metrics.go
+	probeLatency = "hermes_probe_latency_seconds"
+)
 
 // Probe holds aggregate information about all probe runs, per-target.
 // It also holds the config and options used to initialise the probe.
 type Probe struct {
 	name    string
 	config  *probepb.HermesProbeDef
-	targets []*Target
+	targets []*target.Target
 	opts    *options.Options
 	logger  *logger.Logger
 }
 
 // interval returns the probing interval in seconds as a time.Duration.
 // Returns:
 //	- time.Duration: returns the probing interval
-func (mp *Probe) interval() time.Duration {
-	return time.Duration(mp.config.GetIntervalSec()) * time.Second
+func (p *Probe) interval() time.Duration {
+	return time.Duration(p.config.GetIntervalSec()) * time.Second
 }
 
 // timeout returns the probe timeout in seconds as a time.Duration.
 // Returns:
 //	- time.Duration: returns the probe timeout
-func (mp *Probe) timeout() time.Duration {
-	return time.Duration(mp.config.GetTimeoutSec()) * time.Second
+func (p *Probe) timeout() time.Duration {
+	return time.Duration(p.config.GetTimeoutSec()) * time.Second
 }
 
 // Init initializes the probe with the given parameters.
-// This is a required method to implement the cloudprober.Probes.Probe interface.
-func (mp *Probe) Init(name string, opts *options.Options) error {
+// This is a required method to iplement the cloudprober.Probes.Probe interface.
+func (p *Probe) Init(name string, opts *options.Options) error {
 	// Cast from Cloudprober defined interface to HermesProbeDef.
 	// This allows for accessing the variables and methods of a HermesProbeDef object.
 	conf, ok := opts.ProbeConf.(*probepb.HermesProbeDef)
 	if !ok {
 		return fmt.Errorf("invalid argument: opts.ProbeConf is not of type *probepb.HermesProbeDef")
 	}
-	mp.name = name
-	mp.config = conf
+	p.name = name
+	p.config = conf
 
-	for _, t := range mp.config.GetTargets() {
-		mp.targets = append(mp.targets, &Target{
+	for _, t := range p.config.GetTargets() {
+		p.targets = append(p.targets, &target.Target{
 			Target: t,
 			Journal: &journalpb.StateJournal{
 				Intent:    &journalpb.Intent{},
 				Filenames: make(map[int32]string),
 			},
 		})
 	}
-	mp.opts = opts
-	mp.logger = opts.Logger
+	p.opts = opts
+	p.logger = opts.Logger
 
 	return nil
 }
 
 // Start runs the probe indefinitely, unless cancelled, at the configured interval.
 // Probe metrics will be sent via the metricChan at the end of the probe run.
-// This is a required method to implement the cloudprober.Probes.Probe interface.
+// This is a required method to iplement the cloudprober.Probes.Probe interface.
```

Comment:
```suggestion
// This is a required method to implement the cloudprober.Probes.Probe interface.
```

Reply:
I think that the replacement of `mp` with `p` might've backfired in a few places you might've not expected...

Reply:
Yes, very much so 😞